torch/distributed/pipelining/schedules.py

```diff
@@ -1208,47 +1309,84 @@ def _get_1f1b_rank_ops(
         n_local_stages * pp_group_size + 2 * (pp_group_size - 1 - rank)
     ) - (warmup_ops + rank)
 
-    if enable_zero_bubble:
+    if zero_bubble_algorithm is ZeroBubbleAlgorithm.ZB1P:
```

Comment:
could you add a comment on why the post warmups are different between 1p and 2p?